docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,195 @@
+
+* *Define dependencies* - Explicitly declare dependencies in a lock file generated by the package manager. Cachi2 only fetches dependencies that are declared in this file.
+
+* *Reproducible builds* - Cachi2 enforces exact version matching for dependencies, including transitive dependencies. This ensures controlled and reproducible build dependencies.
+
+* *Secure builds* - While package managers use lock files to resolve dependencies automatically, it is crucial to verify the checksums of all the dependencies to safeguard against supply chain attacks.
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have dependencies outside your git repository.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: <package-manager-lock-file> <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `gomod`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip`
+Cachi2 supports pip by parsing *requirements.txt* files placed in the root of your repository and then downloading the specified dependencies. These files should be lockfiles, which include all the transitive dependencies. Each transitive dependency in the *requirements.txt* file must be pinned to a specific version.
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have dependencies outside your git repository.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component’s source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: <package-manager-lock-file> <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `pip`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in the JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
```

Comment:
Specify this to pip. Instead of lockfile, maybe mention the path to the requirements file. @brunoapimentel @chmeliik, is it the path to the directory or the path to the actual file?

Reply:
The `path` would be a path to the top-level directory to a package - the directory which contains pyproject.toml or, for older setuptools-based projects, setup.py or setup.cfg

Cachi2 will automatically look for requirements.txt and requirements-build.txt in this directory, but the user can override this. The whole input might looks like this

```jsonc
{
  "type": "pip",
  // path to the package (relative to the --source directory)
  // defaults to "."
  "path": ".",
  // specify requirements files (relative to the package path)
  // defaults to ["requirements.txt"] or [] if the file does not exist
  "requirements_files": ["requirements.txt", "requirements-extra.txt"],
  // specify *build* requirements files
  // defaults to ["requirements-build.txt"] or [] if the file does not exist
  "requirements_build_files": ["requirements-build.txt"],
}
```
(copied from https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#specifying-packages-to-process)

Reply:
Thanks, Adam.
Have you just made these changes locally? Can you unresolve the suggestion or push the fix?